service/tbc/crawler.go

```diff
@@ -400,14 +406,58 @@ func (s *Server) TxIndexer(ctx context.Context, height, count uint64) error {
 	}
 }
 
-// SyncIndexersToHeight tries to move the various indexers to the suplied
+// SyncIndexersToHeight tries to move the various indexers to the supplied
 // height (inclusive).
 func (s *Server) SyncIndexersToHeight(ctx context.Context, height uint64) error {
+	log.Infof("SyncIndexersToHeight %v", height)
+	defer log.Infof("SyncIndexersToHeight exit")
+
 	log.Tracef("SyncIndexersToHeight")
 	defer log.Tracef("SyncIndexersToHeight exit")
 
+	s.mtx.Lock()
+	if s.indexing {
+		s.mtx.Unlock()
+		return fmt.Errorf("already indexing")
+	}
+	s.indexing = true
+	s.mtx.Unlock()
+
+	defer func() {
+		// unquiesce
+		s.mtx.Lock()
+		s.quiesced = false
+		s.indexing = false
+		// s.clipped = false
+		actualHeight, bhb, err := s.RawBlockHeaderBest(ctx)
+		if err != nil {
+			log.Errorf("sync indexers best: %v", err)
+			s.mtx.Unlock()
+			return
+		}
+		// get a random peer
+		p, err := s.randomPeer(ctx)
+		if err != nil {
+			log.Errorf("sync indexers random peer: %v", err)
+			s.mtx.Unlock()
+			return
+		}
+		s.mtx.Unlock()
+
+		// XXX explain why we need to get more headers here
+		// continue getting headers, XXX this does not belong here either
+		// XXX if bh download fails we will get jammed. We need a queued "must execute this command" added to peer/service.
+		log.Infof("resuming block header download at: %v", actualHeight)
+		err = s.getHeaders(ctx, p, bhb)
+		if err != nil {
```

Comment:
I think this error assignment could be nested:
```suggestion
		if err = s.getHeaders(ctx, p, bhb); err != nil {
```